### PR TITLE
rewrite to be more concise; improve wording

### PR DESCRIPTION
**Pull request by conradsnicta**
@rcurtin 
- rewritten many sentences to have more concise wording
- improved wording for sharper impact
- removed superfluous details
- replaced direct mentions of BERT and random forests with shorter and more general language (see email for more details)

### CODE REVIEW

File: paper.md [hunk 1]
```diff
@@ -39,135 +39,125 @@ bibliography: paper.bib
...
-army knife" for C++-based machine learning [@curtin2013mlpack,
-@curtin2018mlpack].  Its efficient implementations of common and cutting-edge
-machine learning algorithms have been used in a wide variety of scientific and
-industrial applications.  This paper serves to introduce mlpack 4, the newest
-major version of the library, which has been significantly refactored and
-redesigned to allow an easy prototyping-to-deployment pipeline, including
-bindings to other languages (Python, Julia, R, Go, and the command line) that
-allow prototyping to be seamlessly performed in environments other than C++.
+For over 15 years, the mlpack machine learning library has served as a
+``swiss army knife'' for C++-based machine learning [@curtin2013mlpack].
```
Comment (rcurtin) [inline paper.md:46]: The "is it Markdown or LaTeX" question always confuses me with JOSS :smile:

File: paper.md [hunk 2]
```diff
@@ -39,135 +39,125 @@ bibliography: paper.bib
...
-lightweight deployment, making mlpack suitable for a wide range of deployment
+The use of machine learning has become ubiquitous in almost every scientific
+discipline and countless commercial applications [@jordan2015machine,@carleo2019machine].
+There is one important commonality to virtually all of these applications:
+machine learning is often computationally intensive, due to the 
+large number of parameters and large amounts of training data.
+This was the main motivator for the original development of mlpack in the C++
+language, which allows for efficient close-to-the-metal implementations [@curtin2013mlpack].
+
+Development and deployment of applications that use machine learning can also be
```
Comment (rcurtin) [inline paper.md:65]: Just a note here, I'll add back in a simple transition to start this paragraph; I want to approach this section as "okay, everyone always says computational efficiency is important, and we have that covered, but *also* we have the deployment problem that must be addressed, and this deployment problem is generally overlooked, at least until mlpack 4 came along and saved the day for everyone everywhere".  (I might be overstating our contribution there :))

File: paper.md [hunk 3]
```diff
@@ -39,135 +39,125 @@ bibliography: paper.bib
...
-examples of C++ notebooks can be found in the [mlpack examples
-repository](https://github.com/mlpack/examples), and these can easily be run on
-BinderHub.
-
-*New bindings and availability.* Support for the Julia, Go, and R 
+and exhaustive list (including numerous bug fixes and new techniques), the
+`HISTORY.md` file (distributed with mlpack) can be consulted.
+
+*Removed dependencies.*
+In accordance with the vision document [@mlpack2021vision], the majority of the
```
Comment (rcurtin) [inline paper.md:120]: Thanks for catching the missing citation here :+1:

File: paper.md [hunk 4]
```diff
@@ -39,135 +39,125 @@ bibliography: paper.bib
...
+refactoring and redesign work focused on reducing dependencies and compilation overhead.
+This has motivated the replacement of the [Boost](https://www.boost.org)
+C++ libraries, upon which mlpack previously depended upon,
```
Comment (rcurtin) [inline paper.md]:
```suggestion
C++ libraries, upon which mlpack previously depended,
```

File: paper.md [hunk 5]
```diff
@@ -39,135 +39,125 @@ bibliography: paper.bib
...
-# Special Thanks
+# Acknowledgements
 
-mlpack is a community-led library, and the software is the product of the hard
-work of over 220 individuals (at the time of this writing).  We are indebted to
-all of these contributors, as well as all those who have opened issues and bug
-reports over the years.  mlpack's development has been supported by Google, via
-a decade-long participation the Google Summer of Code program, and also by
-NumFOCUS, which fiscally sponsors mlpack.
+Development of mlpack is a community effort. It is the product of hard work by
```
Comment (rcurtin) [inline paper.md]:
```suggestion
Development of mlpack is community-led. It is the product of hard work by
```
I'll make this change to highlight that we are actually, really, truly community-led, not guided by corporate overlords like many other machine learning libraries.  (Not as a knock on those libraries, but I think it is important to differentiate.)

**Review by rcurtin — APPROVED** [review]
Awesome---this is great---thank you!  I have a couple little comments and small changes I'll make after merge, detailed in the review comments. :+1:

File: paper.md (hunk 2 above)
Comment (conradsnicta) [reply paper.md:65]: sounds good
